### PR TITLE
fix: allow cows to be placed on a non-road square

### PR DESCRIPTION
**Pull request by RaihanRasheedApurbo**
## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have linked this PR to a ZenHub Issue.

- - -
This change is [Reviewable](https://reviewable.io/reviews/ocadotechnology/rapid-router/1746)

### CODE REVIEW

File: game/static/game/js/level_editor.js [hunk 1]
```diff
@@ -2972,14 +2962,15 @@ ocargo.LevelEditor = function(levelId) {
...
-                this.valid = true;
-                drawing.setCowImagePosition(coordinates, this.image, this.controlledNode);
+            this.coordinate = new ocargo.Coordinate(data.coordinates[0].x, data.coordinates[0].y);
+            this.valid = isValidDraggedCowPlacement(this.coordinate, this);
+            
+            if (this.isOnRoad()) {
+                const controlledNode = ocargo.Node.findNodeByCoordinate(coordinates, nodes);
+                drawing.setCowImagePosition(this.coordinate, this.image, controlledNode);
+            } else {
+                this.image.transform('...t' + (-paper.scrollLeft())  +  ',' +  paper.scrollTop());
```
Comment (RaihanRasheedApurbo) [inline game/static/game/js/level_editor.js:2973]: @faucomte97 to test these code blocks I need to sign in to the website in my local setup. Can you tell me how to sign up? 

I have manually tested other blocks locally. What else do you suggest me to do to make sure I am not breaking any existing functionalities?

Comment (faucomte97) [reply game/static/game/js/level_editor.js:2973]: Hi @RaihanRasheedApurbo, you can either:
- log in using any of the test teacher accounts already existing in the DB (details of those test accounts can be found here: https://github.com/ocadotechnology/codeforlife-portal/blob/master/portal/migrations/0001_squashed_0041_new_news.py)
- create a user of your own by following the registration process. The content of the emails you'd normally get during the registration process will get outputted to your terminal which is where you'll find the verification link, for example.

Let me know if that makes sense, option 1 is usually the easier one.

Then, regarding your other question - our automated test pipeline here on GitHub will test the functionality of the website, so if anything outside of your test broke, the pipeline will fail. I will approve the pipeline now

**Comment by RaihanRasheedApurbo** [comment]
@faucomte97 I have created an [MR](https://github.com/ocadotechnology/codeforlife-workspace/pull/157) for signing the contributor agreement. I have read the process. My understanding is that I will receive an email if you action something on the [MR](https://github.com/ocadotechnology/codeforlife-workspace/pull/157). 

I think we need to do this first before we can `validate-existing-contributors / validate-existing-contributors ` workflow in this MR.

**Review by faucomte97 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ocadotechnology/rapid-router/1746)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @RaihanRasheedApurbo)

*[a discussion](https://reviewable.io/reviews/ocadotechnology/rapid-router/1746#-OAN14xT1tPO-redGGUQ:-OAN14xT1tPO-redGGUR:b-5n7f4o) (no related file):*
Hi @RaihanRasheedApurbo, I've manually tested this PR and it looks like I'm not able to play a level where I've placed cows on the grass (not on the road).

The map doesn't finish loading and I get this error:



___

**Comment by RaihanRasheedApurbo** [comment]
@faucomte97 can you do another manual test and see if its okay now? Since I haven't worked on this code repo before, I don't know all the potential places things can break. So there is a chance that you might find another bug. I am sorry for this inconvenience.

**Review by faucomte97 — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ocadotechnology/rapid-router/1746#-:-OBBtY0o7uA6QBzzYBRL:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @RaihanRasheedApurbo)

**Pull request merged**